src/CallableResolver/ContainerResolver.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Middlewares\Utils\CallableResolver;
+
+use Closure;
+use Interop\Container\ContainerInterface;
+use RuntimeException;
+
+/**
+ * Resolve a callable using a container.
+ */
+final class ContainerResolver extends Resolver
+{
+    /**
+     * @var ContainerInterface
+     */
+    private $container;
+
+    public function __construct(ContainerInterface $container)
+    {
+        $this->container = $container;
+    }
+
+    public function resolve($callable, array $args = [])
+    {
+        if (is_string($callable)) {
+            $callable = $this->resolveString($callable);
+        }
+
+        if (is_string($callable)) {
+            $callable = $this->container->get($callable);
+
+            if (is_object($callable) && !$callable instanceof Closure) {
+                $callable = [$callable, '__invoke'];
+            }
+        } elseif (is_array($callable) && is_string($callable[0])) {
+            $callable[0] = $this->container->get($callable[0]);
+        } elseif (is_object($callable) && !$callable instanceof Closure) {
+            $callable = [$callable, '__invoke'];
+        }
```

Comment:
Cool. I was a bit confused, because I remembered that way, but I saw some similar things in the code here, so I used it, but I will remove it than.